edx_solutions_api_integration/courses/views.py

```diff
@@ -1930,6 +1922,107 @@ def get(self, request, course_id):  # pylint: disable=W0613,W0221
         return Response(data, status=status.HTTP_200_OK)
 
 
+class CompletionList(SecureListAPIView):  # pylint: disable=too-many-ancestors
+    """
+    ### The CompletionList allows clients to view user's course module completion entities
+    to monitor a user's progression throughout the duration of a course,
+    - URI: ```/api/courses/{course_id}/completions```
+    - GET: Returns a JSON representation of the course, content and user and timestamps
+    - GET Example:
+        {
+            "count":"1",
+            "num_pages": "1",
+            "previous": null
+            "next": null
+            "results": [
+                {
+                    "id": 2,
+                    "user_id": "3",
+                    "course_id": "32fgdf",
+                    "content_id": "324dfgd",
+                    "stage": None,
+                    "created": "2014-06-10T13:14:49.878Z",
+                    "modified": "2014-06-10T13:14:49.914Z"
+                }
+            ]
+        }
+
+    Filters can also be applied
+    ```/api/courses/{course_id}/completions/?user_id={user_id}```
+    ```/api/courses/{course_id}/completions/?content_id={content_id}```
+    ```/api/courses/{course_id}/completions/?user_id={user_id}&content_id={content_id}```
+    - POST: Creates a Course-Module completion entity
+    - POST Example:
+        {
+            "content_id":"i4x://the/content/location",
+            "user_id":4,
+            "stage": "First"
+        }
+    ### Use Cases/Notes:
+    * Use GET operation to retrieve list of course completions by user
+    * Use GET operation to verify user has completed specific course module
+    """
+    serializer_class = BlockCompletionSerializer
+
+    def get_queryset(self):
+        """
+        GET /api/courses/{course_id}/completions/
+        """
+        content_id = self.request.query_params.get('content_id', None)
+        course_id = self.kwargs['course_id']
+        if not course_exists(self.request, self.request.user, course_id):
+            raise Http404
+        course_key = get_course_key(course_id)
+        queryset = BlockCompletion.objects.filter(course_key=course_key).select_related('user')
+        user_ids = get_ids_from_list_param(self.request, 'user_id')
+        if user_ids:
+            queryset = queryset.filter(user__in=user_ids)
+
+        if content_id:
+            content_descriptor, content_key, _existing_content = get_course_child(
+                self.request,
+                self.request.user,
+                course_key,
+                content_id
+            )
+            if not content_descriptor:
+                raise Http404
+            queryset = queryset.filter(block_key=content_key)
+
+        return queryset
+
+    def post(self, request, course_id):
+        """
+        POST /api/courses/{course_id}/completions/
+        """
+        warnings.warn(DeprecationWarning, "BlockCompletions should not be created this way.  Use the suggested APIs instead")
```

Comment:
@jcdyer, the above function call has arguments in the wrong order.

```suggestion
        warnings.warn("BlockCompletions should not be created this way.  Use the suggested APIs instead", DeprecationWarning)
```

Reply:
Yep.  I fixed that one, then fixed it again during the rebase, then it must have broken again later in the rebase.